addons/l10n_ke_edi_tremol/wizard/account_move_send.py

```diff
@@ -0,0 +1,34 @@
+from odoo import _, models, fields, api
+
+
+class AccountMoveSend(models.Model):
+    _inherit = 'account.move.send'
+
+    l10n_ke_edi_warning_message = fields.Text(compute='_compute_l10n_ke_edi_warning_message')
+
+    @api.depends('move_ids')
+    def _compute_l10n_ke_edi_warning_message(self):
+        for wizard in self:
+            warning_moves = wizard.move_ids.filtered(lambda m: m.country_code == 'KE' and not m._l10n_ke_check_invoice_ready())
+            if warning_moves:
+                wizard.l10n_ke_edi_warning_message = '\n'.join([
+                    _("The following documents have no details related to the fiscal device, a proforma invoice will be used."),
+                    *(warning_moves.mapped('name'))
+                ])
+            else:
+                wizard.l10n_ke_edi_warning_message = False
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if invoice.country_code == 'KE' and not invoice._l10n_ke_check_invoice_ready():
```

Comment:
```suggestion
        if not invoice._l10n_ke_check_invoice_ready():
```
the check of the country is already in `_l10n_ke_check_invoice_ready`
Btw, the name of the method doesn't mean anything... what about `_l10n_ke_is_invoice_edi_needed`

Reply:
Yesn't, if the invoice is not from Kenya at all you don't want to set the error/warning.
Maybe it's clearer if I remove the country check from the method ?